### PR TITLE
extension: Reduce code duplication

### PR DESCRIPTION
**Pull request by n0samu**
Avoid concatenating all of `ruffle-core` into the extension's `background.js` and `popup.js` files, significantly reducing their size.
- Mark the `ruffle-core` module as free of side effects, allowing webpack to only include functions that were actually imported.
- Move the `isSwfFilename` and `isSwfMimeType` functions to a separate file, avoiding importing all of `ruffle-player` into the extension's background script.

### CODE REVIEW

**Comment by Dinnerbone** [comment]
How much does this change the size, out of curiosity?

**Comment by Dinnerbone** [comment]
> > How much does this change the size, out of curiosity?
> 
>     * background.js: 158KB => 6KB
> 
>     * popup.js: 162KB => 10KB

Seriously?! That huge? Holy crap.

**Review by Dinnerbone — APPROVED** [review]

**Comment by relrelb** [comment]
If I understand correctly this will make #10422 (and #10454) no longer necessary. Would you like to revert them in this PR?

File: web/packages/core/test/swf-file-name.ts
[the remark this answers is not in the document]
Comment (n0samu) [reply web/packages/core/test/swf-file-name.ts:1]: No, this is a test specifically for the swfFileName function

**Comment by n0samu** [comment]
> If I understand correctly this will make #10422 (and #10454) no longer necessary. Would you like to revert them in this PR?

I think we should still use `import type` where possible as a best practice. I can revert #10454 though since I guess it's no longer important enough to have a custom rule for

**Review by relrelb — APPROVED** [review]

**Pull request merged**